plugins/datadog_tracer/test/samson_datadog_tracer/apm_test.rb

```diff
@@ -0,0 +1,138 @@
+# frozen_string_literal: true
+require_relative "../test_helper"
+
+SingleCov.covered!
+
+describe SamsonDatadogTracer::APM do
+  module DDTracer
+    def self.trace(*)
+      yield
+    end
+  end
+
+  module Datadog
+    def self.tracer
+      DDTracer
+    end
+  end
+
+  describe ".trace_method_execution_scope" do
+    it "skips tracer when disabled" do
+      with_env DATADOG_TRACER: nil do
+        Datadog.expects(:tracer).never
+        SamsonDatadogTracer::APM.trace_method_execution_scope("test") { "without tracer" }
+      end
+    end
+
+    it "trigger tracer when enabled" do
+      with_env DATADOG_TRACER: "1" do
+        Rails.stubs(:env).returns("staging")
+        Datadog.expects(:tracer).returns(DDTracer)
+        SamsonDatadogTracer::APM.trace_method_execution_scope("test") { "with tracer" }
+      end
+    end
+  end
+
+  class TestKlass1
+    include SamsonDatadogTracer::APM
+    SamsonDatadogTracer::APM.module_eval { include Datadog }
+
+    def pub_method
+      :pub
+    end
+
+    trace_method :pub_method
+  end
+
+  describe "skips APM trace methods" do
+    let(:klass) { TestKlass1.new }
+    it "skips tracker when apm is not enabled" do
+      Datadog.expects(:tracer).never
+      klass.send(:pub_method)
+    end
+  end
+
+  ENV.store("DATADOG_TRACER", "1")
```

Comment:
... you ran into a validation that told you why this should not be done and you decided the best cause of action was to hack around it instead of asking for help 🤷‍♂️